dashboard/src/actions/charts.ts

```diff
@@ -90,15 +94,19 @@ export function fetchCharts(
   repos: string,
   page: number,
   size: number,
+  records: Map<number, boolean>,
   query?: string,
 ): ThunkAction<Promise<void>, IStoreState, null, ChartsAction> {
   return async dispatch => {
-    dispatch(requestCharts(query));
-    try {
-      const charts = await Chart.fetchCharts(cluster, namespace, repos, page, size, query);
-      dispatch(receiveCharts(charts));
-    } catch (e) {
-      dispatch(errorChart(new FetchError(e.message)));
+    const lastPendingPage = Array.from(records.keys()).pop() || 1;
+    if (records.get(page) === false && page <= lastPendingPage) {
+      dispatch(requestCharts(page, query));
+      try {
+        const response = await Chart.fetchCharts(cluster, namespace, repos, page, size, query);
+        dispatch(receiveCharts({ items: response.data, page }, response?.meta?.totalPages <= page));
```

Comment:
I would also add the `totalPages` to the payload unless using the meta parameter as you are doing gives some benefit, that way, if in the future you need to add more info, you don't need to guess if it's in one parameter or the other.